index.html

```diff
@@ -215,41 +225,52 @@
 
 //Sort by income and poverty rate
 
+
 function updateTable() {
-  console.log("UpdateTable " + options.filtered)
+
   var data = dc_info.slice()
- if (options.sorted) {
-    if (options.sorted === 'desc') {
-      data.sort(function (a, b) { return b.avg_family_income - a.avg_family_income; });
-    } else if (options.sorted === 'desc-poverty') {
-      data.sort(function (a, b) { return b.poverty_rate - a.poverty_rate; });
-    }
- }
 
 
 //Filter by Poverty>20%
-if (options.filtered) {
+  if (options.filtered) {
      if (options.filtered === 'filter-p') {
         data = data.filter(function (a) { 
           return a.poverty_rate > 20; });
       }
 }
 //Filter by 6 figure income
-if (options.filtered) {
+  if (options.filtered) {
     if (options.filtered === 'filter-i'){
       data = data.filter(function (a) {
         return a.avg_family_income > 99999; });
     }
 }
 
-  
+//Sort
 
+if (options.sorted) {
+    if (options.sorted === 'desc') {
+      data.sort(function (a, b) { return b.avg_family_income - a.avg_family_income; });
+    } else if (options.sorted === 'desc-poverty') {
+      data.sort(function (a, b) { return b.poverty_rate - a.poverty_rate; });
+    }
+ }
 
+//My attempt at Pagination for Next button, not working though
 
-  
 
-//Pagination
 
+  if (options.page) {
+    if (options.page === 'next-page'){
+      
+      data = data.slice(function (pageno, max_num) {
```

Comment:
I think you got twisted around here. You're on the right track, but we're wrapped inside an `Array.slice()` call here, which doesn't work. But if I unwrap it, we're now closer:

```js
  if (options.page) {
    if (options.page === 'next-page') {
      console.log('3');
      var pageno = 2;
      var rows = 10;

      data = data.slice((pageno - 1) * rows, pageno * rows);
    }
  }
```